components/script/dom/globalscope.rs

```diff
@@ -3255,6 +3261,25 @@ impl GlobalScope {
         let _ = self.console_group_stack.borrow_mut().pop();
     }
 
+    pub(crate) fn increment_console_count(&self, label: &DOMString) -> usize {
+        let mut map = self.console_count_map.borrow_mut();
+        if let Some(val) = map.get_mut(label) {
+            *val += 1;
+            return *val;
+        }
+        map.insert(label.clone(), 1);
+        1
+    }
+
+    pub(crate) fn reset_console_count(&self, label: &DOMString) -> Result<(), ()> {
+        let mut map = self.console_count_map.borrow_mut();
+        if let Some(val) = map.get_mut(label) {
+            *val = 0;
+            return Ok(());
+        }
+        return Err(());
+    }
```

Comment:
This would be better served by a match.

```suggestion
        match self.console_count_map.borrow_mut().get_mut(label) {
            Some(value) => {
                *val = 0;
                Ok(())
            }
            None => Err(()),
        }

```